src/client/Microsoft.Identity.Client/Internal/Requests/ClientCredentialRequest.cs

```diff
@@ -193,6 +200,18 @@ private async Task<MsalAccessTokenCacheItem> GetCachedAccessTokenAsync()
 
             if (cachedAccessTokenItem != null && !_clientParameters.ForceRefresh)
             {
+                if (!string.IsNullOrEmpty(_clientParameters.AccessTokenHashToRefresh))
```

Comment:
The [cyclomatic complexity](https://learn.microsoft.com/en-us/visualstudio/code-quality/code-metrics-cyclomatic-complexity?view=vs-2022) of this block of code just exploded. Pls use some private methods to hide some complexity away.